tests/test_bulk_rx_multiep/src/main.xc

```diff
@@ -25,28 +25,22 @@
 XUD_EpType epTypeTableOut[EP_COUNT_OUT] = {XUD_EPTYPE_CTL, XUD_EPTYPE_BUL, XUD_EPTYPE_BUL, XUD_EPTYPE_BUL, XUD_EPTYPE_BUL, XUD_EPTYPE_BUL, XUD_EPTYPE_BUL};
 XUD_EpType epTypeTableIn[EP_COUNT_IN] =   {XUD_EPTYPE_CTL, XUD_EPTYPE_BUL, XUD_EPTYPE_BUL, XUD_EPTYPE_BUL, XUD_EPTYPE_BUL, XUD_EPTYPE_BUL, XUD_EPTYPE_BUL};
 
-int main()
+unsigned test_func(chanend c_ep_out[EP_COUNT_OUT], chanend c_ep_in[EP_COUNT_IN])
 {
-    chan c_ep_out[EP_COUNT_OUT], c_ep_in[EP_COUNT_IN];
-
+    unsigned fail[4];
+    
     par
     {
-        
-        XUD_Main(c_ep_out, EP_COUNT_OUT, c_ep_in, EP_COUNT_IN,
-                                null, epTypeTableOut, epTypeTableIn,
-                                XUD_SPEED_HS, XUD_PWR_BUS);
-
-    
-        TestEp_Rx(c_ep_out[TEST_EP_NUM], TEST_EP_NUM, PACKET_LEN_START, PACKET_LEN_END);
-        TestEp_Rx(c_ep_out[4], 4, PACKET_LEN_START, PACKET_LEN_END);
-        TestEp_Rx(c_ep_out[5], 5, PACKET_LEN_START, PACKET_LEN_END);
-        {
-            XUD_ep ep_out_0 = XUD_InitEp(c_ep_out[0]);
-            TestEp_Rx(c_ep_out[6], 6, PACKET_LEN_START, PACKET_LEN_END);
-            XUD_Kill(ep_out_0);
-            exit(0);
-        }
+        fail[0] = TestEp_Rx(c_ep_out[TEST_EP_NUM], TEST_EP_NUM, PACKET_LEN_START, PACKET_LEN_END);
+        fail[1] = TestEp_Rx(c_ep_out[4], 4, PACKET_LEN_START, PACKET_LEN_END);
+        fail[2] = TestEp_Rx(c_ep_out[5], 5, PACKET_LEN_START, PACKET_LEN_END);
+        fail[3] = TestEp_Rx(c_ep_out[6], 6, PACKET_LEN_START, PACKET_LEN_END);
     }
 
-    return 0;
+    for(size_t i = 0; i< 4; i++)
+        fail[0] |= fail[i];
```

Comment:
How does the engineer investigating a failure tell which run of `TestEp_Rx()` failed?

Reply:
They would have to add debug code. Currently all tests are pass/fail only - no other feedback.

Reply:
I understand, and that's okay with me.

Reply:
I agree that it it would be nice if the test framework did output better failure reasoning.